PIVX-Labs/MyPIVXWallet
# [Refactor] Clean up wallet related code

## Abstract

This is a continuation of the work started in #197 main changes are:

- From now on `Masterkey` has the only role of internally generating addresses and keys without any context, all external information, for example number of generated addresses, map of known addresses, nAccount used in derivation path has been moved to the `Wallet` class
- Having account info only stored in `Wallet` instead of spread in all files is also a first step toward multi account system.
- I  cleaned up almost all files and removed functions that should be called only internally from the Wallet class: for example `getDerivationPath()`  and `getMasterKey()`
- `Wallet` class has been decoupled from GUI
- `Network` class takes now a `Wallet` object instead of a `Masterkey`

Overall, as you can see from the `git diff`, I think that this is a good clean up and from this point we should finally be able to build a good `Wallet` class that will also manage UTXO, masternode, balance...

---

## Testing
Test all features related to wallet:
- Create a new wallet
- Check that you can send and receive funds
- Check that you can generate new addresses
- Check that the contact system works
- Check that you can encrypt the wallet and change password

If any errors are found, the PR works unexpectedly, or you have viable suggestions to improve the UX or functionality of the PR, let me know!

---
## For future contributions
While the clean up is almost completed there are still a few places that can be improved: for example

- In `bitTrx.js` there is still a call to `wallet.getDerivationPath()`
- the `Masternode` class  should also be refactored and incorporated with the new `Wallet` class
